include/z64pause.h

```diff
@@ -0,0 +1,134 @@
+#ifndef Z64PAUSE_H
```

Comment:
Would it be more appropriate to put this header in:
`src/overlays/misc/ovl_kaleido_scope/pause.h`
similar to how there is:
`src/overlays/gamestates/ovl_file_choose/file_select.h`

Reply:
I plan to have this header used for "public" pause stuff that is used in code, and the one in overlays/ for stuff that only matters inside kaleido

It's not always a clear separation but 🤷

Reply:
yea file select gets away with being more isolated cause it is its own gamestate. not so much for pause